### PR TITLE
Bugfix/hw 52146 issue of the table view layout

### PR DESCRIPTION
**Pull request by syyang-hw**

### CODE REVIEW

File: Sources/Extensions/UIImage.swift [hunk 1]
```diff
@@ -24,15 +24,18 @@ extension UIImage {
...
     /// - returns: A string that will appear as icon
-    static func fontIcon(_ name: String, _ textColor: UIColor, _ backgroundColor: UIColor?) -> UIImage {
+    static func fontIcon(_ name: String,
+                         _ iconSize: CGSize,
+                         _ textColor: UIColor,
+                         _ backgroundColor: UIColor?) -> UIImage {
         let paragraph = NSMutableParagraphStyle()
         paragraph.alignment = NSTextAlignment.center
-        let iconSize = CGSize(width: Theme.Icon.size, height: Theme.Icon.size)
         let fontSize = CGFloat(Theme.Icon.size)
```
Comment (gmeyer-hw) [inline Sources/Extensions/UIImage.swift]: We can use the `iconSize` instead of `Theme.Icon.size`.   i.e  `let fontSize = CGFloat(iconSize)`

File: Sources/Generics/GenericTableViewController.swift [hunk 2]
```diff
@@ -60,59 +61,40 @@ class GenericTableViewController<T: GenericCell<ModelType>, ModelType>: UITableV
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.titleDisplayMode(.never)
-        self.extendedLayoutIncludesOpaqueBars = true
+
+        view.backgroundColor = .white
```
Comment (gmeyer-hw) [inline Sources/Generics/GenericTableViewController.swift]: Do we need this `white` color? If yes then you should use `Theme`.

File: Sources/TransferMethod/ListTransferMethodViewController.swift [hunk 3]
```diff
@@ -65,7 +56,10 @@ public final class ListTransferMethodViewController: UITableViewController {
 
     // MARK: - Transfer method list table view dataSource and delegate
     override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return presenter.numberOfCells
+        let numberOfCells = presenter.numberOfCells
+        //viewNoResults.isHidden = presenter.isLoading ? true : numberOfCells > 0
+
+        return numberOfCells
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/ListTransferMethodViewController.swift]: You could revert this change to be this `return presenter.numberOfCells`

File: Sources/TransferMethod/ListTransferMethodViewController.swift [hunk 4]
```diff
@@ -100,6 +94,14 @@ public final class ListTransferMethodViewController: UITableViewController {
         tableView.deselectRow(at: indexPath, animated: true)
     }
 
+    override public func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        return nil
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/ListTransferMethodViewController.swift]: We could remove this implementation if the return value is `nil`

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 5]
```diff
@@ -28,14 +28,9 @@ struct SelectTransferMethodTypeConfiguration {
 final class SelectTransferMethodTypeCell: UITableViewCell {
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: Suggestion:  The `ListTransferMethodTableViewCell ` and `SelectTransferMethodTypeCell ` except for the `configure` in the `detailTextLabel?.attributedText = ...` part.  Maybe we could create a common `UITableViewCell`

File: Sources/Extensions/UIImage.swift (hunk 1 above)
Comment (syyang-hw) [reply Sources/Extensions/UIImage.swift]: Sure

File: Sources/Generics/GenericTableViewController.swift (hunk 2 above)
Comment (syyang-hw) [reply Sources/Generics/GenericTableViewController.swift]: No need to add this color, copied from previous pr

File: Sources/TransferMethod/ListTransferMethodViewController.swift (hunk 3 above)
Comment (syyang-hw) [reply Sources/TransferMethod/ListTransferMethodViewController.swift]: Yea, same here copied from another pr

File: Sources/TransferMethod/ListTransferMethodViewController.swift (hunk 4 above)
Comment (syyang-hw) [reply Sources/TransferMethod/ListTransferMethodViewController.swift]: We do need this method because this table is grouped. There will be some margin between the cell and the navigation bar if we do not have this method.

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 5 above)
Comment (syyang-hw) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: We can create another ticket for it.

File: Sources/Extensions/String+Array.swift [hunk 6]
```diff
@@ -44,6 +44,25 @@ extension String {
     }
 }
 
+extension NSMutableAttributedString {
```
Comment (gmeyer-hw) [inline Sources/Extensions/String+Array.swift]: Suggestion: ``` swift 
extension NSMutableAttributedString {
    func append(value: String, font: UIFont, color: UIColor) {
        append(value: value,attributes:  [
                .font: font,
                .foregroundColor: color
            ])
    }

    func append(value: String, color: UIColor) {
        append(value: value,attributes: [.foregroundColor: color] )
    }

    func append(value: String, attributes: [NSAttributedString.Key: Any]?) {
        append(
            NSAttributedString(string: value,
                               attributes: attributes))
    }
}
```

File: Sources/Helper/Theme.swift [hunk 7]
```diff
@@ -101,8 +101,12 @@ public struct Theme {
 
     /// Representation of all customizable visual style property for the `Hyperwallet`'s icon.
     public struct Icon {
-        /// The icon size
+        /// The icon font size
         public static let size = 20
+        /// The icon image height
+        public static let height = 40
```
Comment (gmeyer-hw) [inline Sources/Helper/Theme.swift]: Suggestion: The image is always a square, so you could replace these two variables (`height` and  `width`) to `frameDimention` or `frameSize` = 40.

File: Sources/TransferMethod/ListTransferMethodTableViewCell.swift [hunk 8]
```diff
@@ -41,43 +37,65 @@ final class ListTransferMethodTableViewCell: UITableViewCell {
...
 extension ListTransferMethodTableViewCell {
     func configure(configuration: ListTransferMethodCellConfiguration) {
-        transferMethodTitle.text = configuration.transferMethodType
-        transferMethodSubTitleOne.text = configuration.transferMethodCountry
-        transferMethodSubTitleTwo.text = configuration.transferMethodExpiryDate
-        iconView.draw(fontName: configuration.transferMethodIconFont)
+        textLabel?.text = configuration.transferMethodType
+        detailTextLabel?.attributedText = formatSubtitle(transferMethodCountry: configuration.transferMethodCountry,
+                                                         lastFourDigitAccountNumber: configuration
+                                                            .lastFourDigitAccountNumber)
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/ListTransferMethodTableViewCell.swift]: We don't need to use `attributedText` in this case because of the font and color are simple. 
Suggestion: 
- Remove this function `formatSubtitle` 
- just concatenate the country and the last four digit account number

``` swift 
        var detailText = String(format: "%@\n", configuration.transferMethodCountry)

        if let lastFourDigitAccountNumber = configuration.lastFourDigitAccountNumber {
            detailText = String(format: "%@%@", detailText, lastFourDigitAccountNumber)
        }

        detailTextLabel?.text = detailText
```

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 9]
```diff
@@ -44,47 +39,52 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
...
     }
 
     @objc dynamic var titleLabelFont: UIFont! {
-        get { return self.titleLabel.font }
-        set { self.titleLabel.font = newValue }
+        get { return self.textLabel?.font }
+        set { self.textLabel?.font = newValue }
     }
 
     @objc dynamic var descriptionLabelColor: UIColor! {
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: We are not using this property. Could you remove?

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift [hunk 10]
```diff
@@ -44,47 +39,52 @@ final class SelectTransferMethodTypeCell: UITableViewCell {
...
-        get { return self.descriptionLabel.textColor }
-        set { self.descriptionLabel.textColor = newValue }
+        get { return self.detailTextLabel?.textColor }
+        set { self.detailTextLabel?.textColor = newValue }
     }
 
     @objc dynamic var descriptionLabelFont: UIFont! {
```
Comment (gmeyer-hw) [inline Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: We are not using this property. Could you remove

File: Sources/Extensions/String+Array.swift (hunk 6 above)
Comment (syyang-hw) [reply Sources/Extensions/String+Array.swift]: What's the purpose of having this? ``` func append(value: String, attributes: [NSAttributedString.Key: Any]?) {
        append(
            NSAttributedString(string: value,
                               attributes: attributes))
    }``` it's just a wrapper and the logic is exactly the same as the ```append``` method swift provides us.

File: Sources/Helper/Theme.swift (hunk 7 above)
Comment (syyang-hw) [reply Sources/Helper/Theme.swift]: Yes, good idea

File: Sources/TransferMethod/ListTransferMethodTableViewCell.swift (hunk 8 above)
Comment (syyang-hw) [reply Sources/TransferMethod/ListTransferMethodTableViewCell.swift]: We are going to create a ticket to have a common ```configuration``` class to deal with both list cell and select cell. This method will be changed as well. Not in this pr

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 9 above)
Comment (syyang-hw) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: Sure

File: Sources/TransferMethod/SelectTransferMethodTypeCell.swift (hunk 10 above)
Comment (syyang-hw) [reply Sources/TransferMethod/SelectTransferMethodTypeCell.swift]: Added this in theme manager
